porter/v1/compliance.proto

```diff
@@ -0,0 +1,39 @@
+syntax = "proto3";
+
+package porter.v1;
+
+enum EnumComplianceVendor {
+  ENUM_COMPLIANCE_VENDOR_UNSPECIFIED = 0;
+  // ENUM_COMPLIANCE_VENDOR_VANTA signifies the compliance vendor Vanta.
+  ENUM_COMPLIANCE_VENDOR_VANTA = 1;
+}
+
+enum EnumComplianceCheckStatus {
+  ENUM_COMPLIANCE_CHECK_STATUS_UNSPECIFIED = 0;
+  // ENUM_COMPLIANCE_CHECK_STATUS_PASSED signifies the compliance check passed.
+  ENUM_COMPLIANCE_CHECK_STATUS_PASSED = 1;
+  // ENUM_COMPLIANCE_CHECK_STATUS_FAILED signifies the compliance check failed.
+  ENUM_COMPLIANCE_CHECK_STATUS_FAILED = 2;
+  // ENUM_COMPLIANCE_CHECK_STATUS_NOT_APPLICABLE signifies the compliance check was not applicable.
+  ENUM_COMPLIANCE_CHECK_STATUS_NOT_APPLICABLE = 3;
+}
+
+message ContractComplianceCheckGroup {
+  // name is the name of the compliance check group.
+  string name = 1;
+  // status is the current status of the compliance check group.
+  EnumComplianceCheckStatus status = 2;
+  // message is an optional message indicating why the compliance check group is in the current state.
+  string message = 3;
+}
+
+message VendorComplianceCheck {
```

Comment:
If I understand this correctly, a check group is made up of a bunch of vendor compliance checks.  Why then doesn't the ContractComplianceCheckGroup have as a subfield a list of VendorComplianceChecks?

Reply:
No the status of a vendor check is derived based on the status of a given check group, which are the internal checks that porter runs according to fields set in the api contract.